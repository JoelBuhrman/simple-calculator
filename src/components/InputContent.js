import React from 'react'

const InputContent = ({ label, value, setValue, calculate }) => {
  const updateValue = ({ target }) => {
    const { value } = target
    setValue(parseInt(value, 10))
    calculate()
  }

  return (
    <div>
      {label}:
      <input type='number' value={value} onChange={updateValue} />
    </div>
  )
}

export default InputContent
