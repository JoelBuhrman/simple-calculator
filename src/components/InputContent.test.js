import React from 'react'
import TestRenderer from 'react-test-renderer'
import InputContent from './InputContent'

it('renders without crashing', () => {
  const snapshot = TestRenderer.create(
    <InputContent
      label={'Value 1'}
      value={0}
      setValue={jest.fn()}
      calculate={jest.fn()}
    />
  ).toJSON()
  expect(snapshot).toMatchSnapshot()
})
