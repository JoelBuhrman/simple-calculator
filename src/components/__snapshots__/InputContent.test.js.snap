// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  Value 1
  :
  <input
    onChange={[Function]}
    type="number"
    value="0"
  />
</div>
`;
